SilverSolver/assignment2/binary_tree.py

```diff
@@ -0,0 +1,61 @@
+class BinaryNode:
+
+    def __init__(self, value, parent=None, child1=None, child2=None):
+        self.value  = value
+        self.parent = parent
+        self.child1 = child1
+        self.child2 = child2
+
+    def get_ancestors(self, prev_ancestors=None, value=True):
```

Comment:
I see you are supporting both returning the values and returning the nodes. The exercise is not saying explicitly which way we should go, so choose one direction, and document it. This will simplify the code a bit.
Myself, I went with returning the *values* in my solution.

Reply:
I'm not sure what do you mean by "document it". Also look to my answer to your first comment about why I need to supporting the both returning the values and the objects.

Reply:
Sorry, I do not see how the answer to the first comment explains the value of supporting two types of return values. Polymorphic methods **on the return value** are quite uncommon. In the context of an inteview, where the whiteboards are not always big, and the time is really limited. I would focus on the minimum amount of code to solve the problem, not adding extra code for extra features.